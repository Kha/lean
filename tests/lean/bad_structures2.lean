structure foo :=
(x : bool)

structure boo :=
(x : nat)

structure bla extends foo, boo

structure boo2 :=
{x : bool}

structure bla extends foo, boo2

structure bla extends foo :=
(x : num)

structure bla extends foo :=
( : num)

structure bla extends foo :=
mk :: y z : num

structure bla2 extends nat

structure bla2 extends Type


structure bla2 : Prop :=
(x : Prop)

structure bla3 : Prop :=
(x : nat)
