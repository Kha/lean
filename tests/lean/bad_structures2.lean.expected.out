bad_structures2.lean:7:27: error: invalid 'structure' header, field 'x' from 'boo' has already been declared
bad_structures2.lean:12:27: error: invalid 'structure' header, field 'x' from 'boo2' has already been declared
bad_structures2.lean:15:1: error: field 'x' has been declared in parent structure
bad_structures2.lean:18:2: error: invalid field, identifier expected
bad_structures2.lean:23:23: error: invalid 'structure' extends, 'nat' is not a structure
bad_structures2.lean:25:23: error: invalid 'structure', expression must be a 'parent' structure
bad_structures2.lean:28:0: error: failed to generate projection 'bla2.x' for 'bla2', type is an inductive predicate, but field is not a proposition
bad_structures2.lean:31:0: error: failed to generate projection 'bla3.x' for 'bla3', type is an inductive predicate, but field is not a proposition
